src/lib/game/world/map.js

```diff
@@ -3,20 +3,31 @@ import THREE from 'three';
 import ADT from '../../pipeline/adt';
 import Chunk from '../../pipeline/adt/chunk';
 import DBC from '../../pipeline/dbc';
-import M2 from '../../pipeline/m2';
 import WDT from '../../pipeline/wdt';
-import WMO from '../../pipeline/wmo';
+import DoodadManager from './doodad-manager';
+import WMOManager from './wmo-manager';
+import TerrainManager from './terrain-manager';
 
 class WorldMap extends THREE.Group {
 
   static ZEROPOINT = ADT.SIZE * 32;
 
   static CHUNKS_PER_ROW = 64 * 16;
-  static CHUNK_RENDER_RADIUS = 10;
+
+  // Controls when ADT chunks are loaded and unloaded from the map.
+  static CHUNK_RENDER_RADIUS = 12;
 
   constructor(data, wdt) {
     super();
 
+    this.matrixAutoUpdate = false;
+
+    this.terrainManager = new TerrainManager(this);
+    this.doodadManager = new DoodadManager(this);
+    this.wmoManager = new WMOManager(this);
+
+    window.wmoManager = this.wmoManager;
```

Comment:
Woops! I think a few things like that might have snuck in. Was using it to check on values when testing. :)